File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.LinkedList;
import java.util.List;
import java.util.PriorityQueue;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
	public static int[][] graph;

	public static int n;
	public static int m;
	public static int v;
	public static boolean[] visited_dfs;
	public static boolean[] visited_bfs;

	public static Queue<Integer> queue;

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());

		n = Integer.parseInt(st.nextToken());
		m = Integer.parseInt(st.nextToken());
		v = Integer.parseInt(st.nextToken());

		visited_dfs = new boolean[n+1];
		visited_bfs = new boolean[n+1];
		graph = new int[n+1][n+1];

		for(int i = 0;i<m;i++){
			String [] tmp = br.readLine().split(" ");
			int start = Integer.parseInt(tmp[0]);
			int end = Integer.parseInt(tmp[1]);

			graph[start][end] = 1;
			graph[end][start] = 1;
		}

		DFS(v);
		System.out.println();
		BFS(v);
	}

	public static void DFS(int source) {
		// int start = source;
		visited_dfs[source] = true;
		System.out.print(source + " ");

		for(int i = 1 ; i< n+1;i++){
			if(graph[source][i] == 1 && !visited_dfs[i]){
				DFS(i);
			}
		}
	}

	public static void BFS(int source){
		queue = new LinkedList<>();
		queue.add(source);

		while(!queue.isEmpty()){
			int num = queue.poll();
			System.out.print(num+" ");
			visited_bfs[num] = true;
			for(int i = 1; i<n+1;i++){
				if(graph[num][i] == 1 && !visited_bfs[i]){
					visited_bfs[i] =true;
					queue.add(i);
				}
			}
		}
	}
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys

def input():
    return sys.stdin.readline().rstrip()

N, M, V = map(int,input().split())

graph = [[] for _ in range(N+1)]
visited_dfs = [False for _ in range(N+1)]
visited_bfs = [False for _ in range(N+1)]

result_dfs = []
result_bfs = []

for _ in range(M):
    start, end = map(int,input().split())
    graph[start].append(end)
    graph[end].append(start)

def dfs(v):
    visited_dfs[v] = True
    print(v, end=' ')

    for i in range(1,N+1):
        if not visited_dfs[i] and i in graph[v]:
            dfs(i)


def bfs(v):
    queue = [v]
    visited_bfs[v] = True

    while queue:
        v = queue.pop(0)
        print(v, end=' ')

        for i in range(1,N+1):
            if not visited_bfs[i] and i in graph[v]:
                queue.append(i)
                visited_bfs[i] = True

dfs(V)
print()
bfs(V)